test/internal/datapath/datapath-win.go

```diff
@@ -0,0 +1,173 @@
+	}
+	logrus.Infof("Second pod: %v %v", secondPod.Name, secondPod.Status.PodIP)
+
+	// Ping the second pod from the first pod located on different nodes
+	return podTest(ctx, clientset, firstPod, []string{"ping", secondPod.Status.PodIP}, rc, pingPassedWindows)
+}
+
+func WindowsPodToNode(ctx context.Context, clientset *kubernetes.Clientset, nodeName, nodeIP, podNamespace, labelSelector string, rc *restclient.Config) error {
+	logrus.Infof("Get Pods by Node: %s", nodeName)
+	pods, err := k8sutils.GetPodsByNode(ctx, clientset, podNamespace, labelSelector, nodeName)
+	if err != nil {
+		logrus.Fatalf("could not get k8s clientset: %v", err)
+	}
+	if len(pods.Items) <= 1 {
+		return errors.New("Less than 2 pods on node")
+	}
+	// Get first pod on this node
+	firstPod, err := clientset.CoreV1().Pods(podNamespace).Get(ctx, pods.Items[0].Name, metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("Getting pod %s failed with %v", firstPod.Name, err))
+	}
+	logrus.Infof("First pod: %v", firstPod.Name)
+
+	// Get the second pod on this node
+	secondPod, err := clientset.CoreV1().Pods(podNamespace).Get(ctx, pods.Items[1].Name, metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("Getting pod %s failed with %v", secondPod.Name, err))
+	}
+	logrus.Infof("Second pod: %v", secondPod.Name)
+
+	// Ping from pod to node
+	resultOne := podTest(ctx, clientset, firstPod, []string{"ping", nodeIP}, rc, pingPassedWindows)
+	resultTwo := podTest(ctx, clientset, secondPod, []string{"ping", nodeIP}, rc, pingPassedWindows)
+
+	if resultOne != nil {
+		return resultOne
+	}
+
+	if resultTwo != nil {
+		return resultTwo
+	}
+
+	return nil
+}
+
+func WindowsPodToInternet(ctx context.Context, clientset *kubernetes.Clientset, nodeName, podNamespace, labelSelector string, rc *restclient.Config) error {
+	logrus.Infof("Get Pods by Node: %s", nodeName)
+	pods, err := k8sutils.GetPodsByNode(ctx, clientset, podNamespace, labelSelector, nodeName)
+	if err != nil {
+		logrus.Fatalf("could not get k8s clientset: %v", err)
+	}
+	if len(pods.Items) <= 1 {
+		return errors.New("Less than 2 pods on node")
+	}
+
+	// Get first pod on this node
+	firstPod, err := clientset.CoreV1().Pods(podNamespace).Get(ctx, pods.Items[0].Name, metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("Getting pod %s failed with %v", firstPod.Name, err))
+	}
+	logrus.Infof("First pod: %v", firstPod.Name)
+
+	// Get the second pod on this node
+	secondPod, err := clientset.CoreV1().Pods(podNamespace).Get(ctx, pods.Items[1].Name, metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("Getting pod %s failed with %v", secondPod.Name, err))
+	}
+	logrus.Infof("Second pod: %v", secondPod.Name)
+
+	resultOne := podTest(ctx, clientset, firstPod, []string{"powershell", "Invoke-WebRequest", "www.bing.com", "-UseBasicParsing"}, rc, invokeWebRequestPassedWindows)
```

Comment:
Any reason we used `Invoke-WebRequest` instead of `Invoke-RestMethod` ? 
I think we can use json output in the `Invoke-RestMethod` case to filter the response instead of parsing the string of status check.

Reply:
After looking through Invoke-RestMethod and the json that is generated, you cannot pull any status code from the json. As of powershell 7.X you can use a parameter to hold the status code `-StatusCodeVariable <String>`, but 5.1 does not have this feature. Invoke-WebRequest can be used to receive the status code in one command, but you lose the context if the command fails as the output changes from the entire Get request to just a status code.

I believe we can get by with just the status code, but seeing as this is being used for testing we need to ensure a verbose output on fail.